autoload/vimcomplete/buffer.vim

```diff
@@ -162,7 +162,7 @@ def CurBufMatches(prefix: string): list<dict<any>>
         endtry
         while [lnum, cnum] != [0, 0]
             var [endl, endc] = icasepat->searchpos('ceW') # end of matching string
-            var mstr = getline(lnum)->strpart(cnum - 1, endc - cnum + 1)
+            var mstr = getline(lnum)->matchstr($'\%{cnum}c.*\%{endc}c.')
```

Comment:
I think what is happening is that `searchpos` returns byte index, and not char index. `cnum` and `endc` are byte indices. These have to be converted to char indices. Can you try `strcharpart(...)`? There is also `charidx()`. Are these characters utf-8?

Reply:
After some testing, I find that using `charidx()` / `strcharpart()` here will get confusing result for CJK and non-CJK.

```vim
vim9script

const data =<< EOF
stmt: hello, 世界!
EOF

def GetCnum(pattern: string): number
    cursor(1, 1)
    return pattern->searchpos('W')[1]
enddef

def GetEndc(pattern: string): number
    cursor(1, 1)
    return pattern->searchpos('ceW')[1]
enddef

const line = data[0]
for pattern in ['hello', '世界']
    # "cursor(1, 1)" in GetCnum / GetEndc ensures that search begins from file top,
    # so we get consistent result.
    const cnum = GetCnum(pattern)
    const endc = GetEndc(pattern)
    echo cnum endc
    echo line->strcharpart(line->charidx(cnum) - 1, line->charidx(endc) - line->charidx(cnum) + 1)
endfor

# result (expect "世界", got " 世"):
#
# 7 11
# hello
# 14 17
#  世
```

Reply:
Moving `XXX - 1` into `charidx()` solves problem:

```vim
    echo line->strcharpart(line->charidx(cnum - 1), line->charidx(endc - 1) - line->charidx(cnum - 1) + 1)
```

I will update commit accordingly.

Reply:
`cnum` / `endc` is NOT zero-based, so before passing to `strcharpart()` / `charidx()`, they needs to minus 1.